### PR TITLE
Implement test_add_str for int and test_add_bool for string

### PR DESCRIPTION
**Pull request by winza83**
Attempt no. 3 - cleaned code, removed additional unexpected successes and implemented isinstance() and datatypes() functions - tests passed for test_add_bool for string and test_add_int for string and test_add_str for int.

### CODE REVIEW

File: batavia/utils.js [hunk 1]
```diff
@@ -468,21 +494,26 @@ batavia.operators = {
...
     },
     SUBTRACT: function(a, b) {
-        if (typeof a === 'string') {
-            if (typeof b === 'string') {
-                throw new batavia.builtins.TypeError("unsupported operand type(s) for -: 'str' and 'str'");
-            }
-        } else {
-             return a - b;
-        }
+        return a - b;
```
Comment (freakboy3742) [inline batavia/utils.js]: This looks like a merge problem - one that explains why the test_subtract_str test needed to be added back in.

File: tests/datatypes/test_str.py [hunk 2]
```diff
@@ -261,6 +259,7 @@ class BinaryStrOperationTests(BinaryOperationTestCase, TranspileTestCase):
         'test_subtract_list',
         'test_subtract_none',
         'test_subtract_set',
+        'test_subtract_str',
         'test_subtract_tuple',
```
Comment (freakboy3742) [inline tests/datatypes/test_str.py]: This change won't be needed if you correct the merge problem I flagged above.

**Comment by freakboy3742** [comment]
@winza83 Looks like you're getting *really* close - but not quite there yet. CI is showing one failure:

    test_add_str (tests.datatypes.test_int.InplaceIntOperationTests)

I've commented on the specific line that is a problem - put that line back in, and this problem will be resolved. There's also one unexpected success; Line 3418 of the CI log is:

    test_add_str (tests.datatypes.test_bool.BinaryBoolOperationTests) ... unexpected success

So, one of the test_bool tests is unexpectedly passing as well. Remove the line tracking this test from the test_bool `not_implemented` list, and you'll have a clean CI check!

There's also one problem that looks like it's been caused by a merge problem - it looks like you've inadvertently removed the implementation of SUBTRACT; as a result, you've have to add back in a test to the `not_implemented` list for `test_subtract_str`. If you restore the old SUBTRACT implementation, you should find that test passes again.

One last thing - there's no need to close this PR and open a new one; you can just commit a fix and the existing PR will be updated. Don't worry about the "noise" of lots of little commits - I can clean that up when I accept the PR.

You're making really good progress! These are all really minor issues - I promise you're just a handful of keystrokes away from a merged pull request!

**Comment by winza83** [comment]
@freakboy3742  thanks for the feedback.  I've made some changes and had sone some commits.  Not sure if CI will run automatically but hopefully this will warrant a successful merge.

**Comment by freakboy3742** [comment]
Due to some amazingly bad timing, your PR collided with one submitted yesterday (#67), which would have left you with yet another complex merge to manage. You've been working so diligently on this, and you've already had to manage a couple of collisions; it seemed unfair to ask you to do *another* merge. So, I've given you an assist and done the merge for you.

With that: Congratulations! You have your first code commit in Batavia!

**Pull request merged**